dts/bindings/comparator/nordic,nrf-comp.yaml

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Nordic nRF COMP (analog COMParator)
+
+  This comparator has varying configurations which require varying
+  properties be set in the devicetree.
+
+  The following example displays the minimum node layout:
+
+    comp: comp@deadbeef {
+            compatible = "nordic,nrf-comp";
+            reg = <0xdeadbeef 0x1000>;
+            interrupts = <0 NRF_DEFAULT_IRQ_PRIORITY>;
+            status = "disabled";
+    };
+
+  Enabling the comparator node requires setting the default
+  configuration of the comparator.
+
+  The following example displays enabling the comparator in
+  single-ended mode, selecting an internal reference:
+
+    &comp {
+            status = "okay";
+            positive-input = "INPUT_0";
+            reference-selection = "INT_1V2";
+            main-mode = "SE";
+            speed-mode = "NORMAL";
+            threshold-up = <36>;
+            threshold-down = <30>;
+            isource = "OFF";
+    };
+
+  To select an external reference, select the "AREF"
+  reference and add the external reference:
+
+    &comp {
+            ...
+            reference-selection = "AREF";
+            external-reference = "EXT_REF_1";
+            ...
+    };
+
+  The following example displays enabling the comparator
+  in differential mode:
+
+    &comp {
+            status = "okay";
+            positive-input = "INPUT_0";
+            external-reference = "EXT_REF_1";
+            main-mode = "DIFF";
+            speed-mode = "NORMAL";
+            hysteresis = "50MV";
+            isource = "OFF";
+    };
+
+compatible: "nordic,nrf-comp"
+
+include: base.yaml
+
+properties:
+  positive-input:
+    type: string
+    enum:
+      - "INPUT_0"
```

Comment:
Because this matches what the NRFX team has decided to use, which I'm trying to follow as closely as possible, on the presumption that there is likely a reason why they chose to use `INPUT_0` for example :)

Reply:
Done, now using AINx for both `psel` and `extrefsel`